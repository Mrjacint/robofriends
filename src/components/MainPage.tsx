import React, { useEffect } from "react";

import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundry from "./ErrorBoundry";
import Header from "./Header";

import "./MainPage.css";
import { IRobot, ImainProps } from '../types'



const MainPage: React.FunctionComponent<ImainProps> = (props) => {
  const {
    onRequestRobot,
    robots,
    searchField,
    onSearchChange,
    isPending,
  } = props;

  console.log(props)

  useEffect(() => {
    onRequestRobot();
  }, [onRequestRobot]);

  const filteredRobots: IRobot[] = robots.filter((robot: IRobot) => {
    return robot.name.toLowerCase().includes(searchField.toLowerCase());
  });

  return isPending ? (
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <Header />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
};

export default MainPage;
