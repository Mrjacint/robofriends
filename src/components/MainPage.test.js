import React from "react";
import { shallow } from "enzyme";

import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobot: jest.fn(),
    robots: [
      {
        id: 3,
        name: "jhon",
        email: "jhon@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("expect to render MainPage component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("filters robots correctly", () => {
  //console.log(wrapper.find("CardList").props().robots);
  const mockProps2 = {
    onRequestRobot: jest.fn(),
    robots: [
      {
        id: 3,
        name: "jhon",
        email: "jhon@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);

  expect(wrapper.find("CardList").props().robots).toEqual([]);
  expect(wrapper2.find("CardList").props().robots).toEqual([]);
});

it("filters robots correctly 2", () => {
  const mockProps2 = {
    onRequestRobot: jest.fn(),
    robots: [
      {
        id: 3,
        name: "jhon",
        email: "jhon@gmail.com",
      },
    ],
    searchField: "jhon",
    isPending: false,
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);

  expect(wrapper2.find("CardList").props().robots).toEqual([
    {
      id: 3,
      name: "jhon",
      email: "jhon@gmail.com",
    },
  ]);
});

it("expect render all components", () => {
  //console.log(wrapper.debug());
  expect(wrapper.find("div").length).toBe(1);
});

it("expect render loading text", () => {
  //console.log(wrapper.debug());
  wrapper.setProps({ isPending: true });
  expect(wrapper.find("div").length).toBe(0);
  expect(wrapper.find("h1").length).toBe(1);
  expect(wrapper.find("h1").text()).toContain("Loading");
});
