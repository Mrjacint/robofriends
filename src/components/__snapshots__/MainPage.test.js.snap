// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
